### DISCUSSION TITLE
Do you use Dexcom One with a transmitter serial number that starts with C?

### DISCUSSION THREAD

**Discussion by Navid200** [NightscoutFoundation/xDrip#2780] [General]
Please proceed only if you are using xDrip as a collector, not a follower using the Dexcom app.

In that case, please make sure you are using the correct settings:
https://navid200.github.io/xDrip/docs/G6-Recommended-Settings.html

Then, try any release after March 6, 2023.  If it works or fails, let me know.   

Please post here and let me know which release of xDrip works for you and which fails.
Also include your phone brand, model and Android version on it.  
This may let us see clues for example what is common between the problem cases.  Or, what is common for success cases.  

Please don't use this thread to ask for help.  If you need help, please start a new thread.

| User | xDrip release in use | xDrip release that fails | Phone | Android |  
| ----- | --------------------- | ------------------------- | ------- | -------- |  
| @hanbozdugan  | 2023.04.13 | | Xiaomi | 12 |  
| @MarkoJ1974 | 2023.04.13 | | Samsung S22 | 13 |  
| @coriangelescu | 2023.03.06 | | Samsung s10 | 11 |  
| @mirata6 | 2023.01.12 | March | Samsung A52 | 13 |  
| @haniotij | 2023.04.13 | | Samsung | 13 |  
| @marinabjezancevic | 2023.03.28 | | Samsung | 12 |

**Comment by hanbozdugan** [comment]
I have a Xiaomi Redmi Note 9 Pro phone updated with Android 12. I am using a Dexcom One sensor with a transmitter starting with the letter C... I have installed the latest version of xDrip+ which is from 13/04/2023 and everything works perfectly. Even now, the sensor I'm using is already 17 days old and still shows accurately. My xDrip+ settings are exactly the same as this documentation https://navid200.github.io/xDrip/docs/G6-Recommended-Settings.html . However, I have noticed that a friend of mine with a different brand of phone had complained that the data coming from the transmitter was being lost when the phone was locked. I think his phone was an LG. After we activated in G5/G6/Dex1 Debug Settings -> Special Pairing Workaround and restarted his phone, he said that it was already OK with him. For me, both with the option off and with the xDrip+ option working fine. But now I'm testing with the option turned on how xDrip+ will behave going forward. Everything is OK at the moment.

**Comment by MarkoJ1974** [comment]
Three days ago I started using DexcomOne serial starting with a "C" and for now it it working great. Had some pairing problems in the first day but I changed some settings and now it works OK. I am using a Samsung S22 Ultra with latest Android 13 and latest Xdrip Nighty

**Comment by coriangelescu** [comment]
I am using a dexcom one transmitter starting with C since 35 days, with the dexcom version from 2023.03.06, on a samsung s10 with Android 11, version 3.1. and it works perfect for me. I am în the 17-th day of the sensor and calibrating once a day it has a max difference of 20 mg/dl. No lost signal, no pairing problems. Therefore, I am afraid to update to a newer dexcom version. Hope it is useful

**Comment by mirata6** [comment]
My son has a dexcom 1 sensor starting with C, at first everything was fine, but after the Xdrip+ updates with versions from March, I could not even once connect and work!  My son has a Samsung A52s Android 13, the version from 12.01.2023 works relatively well with us!

**Comment by haniotij** [comment]
I am using the latest version of xdrip (April  13) with a samsung 21 ultra - ONE UI 5.1 android version 13, as a collector.  I have no problems so far. I will check it again when it is time to place a new sensor. I had previously January's xdrip version. I jumped to the new version (version April 13) on April 14. I am using dexcom one starting with C.

**Comment by marinabjezancevic** [comment]
I am using Dexcom transmitter serial number starting with C. 
First on Samsung Galaxy S9, Android version 9, then switched to a newer phone - Samsung Galaxy A13, Android version 12. 
Xdrip release 28.03.2023. 
Works fine. 
Considering issues that other people complained, I didn't try any newer releases.

**Reply by Navid200** [reply]
The major change affecting this particular transmitter was the release on March 15, 2023.
If you are using the March 28, 2023 release and it is working fine, you should have no trouble using any other releases after that.

**Reply by marinabjezancevic** [reply]
Edit! I started having connectivity issues with the newer phone. Tried release 6h March 2023, also keeps disconnecting.

**Reply by Navid200** [reply]
Do you have special pairing workaround on G5/G6/Dex1 Debug settings page enabled?
Do you have trust auto-connect on the Bluetooh settings page disabled?
Do you have wake workaround on the other misc options page enabled?

**Reply by marinabjezancevic** [reply]
Xdrip settings okay. The bluetooth one initialy I could not find, now found it under location settings and disabled it. 
Will let you know if it fixed the issue.

**Reply by marinabjezancevic** [reply]
So far working good!

**Reply by marinabjezancevic** [reply]
All settings okay, disconnected in the middle of the night again!..

**Reply by Navid200** [reply]
Let's keep an eye on it and see what the average rate of disconnect is.
There was a suggestion on facebook by one or two individuals that something happened to xDrip in the March 15, 2023 release resulting in that release and all the following to have serious connectivity issues.

That's why I opened this thread to gather data and see if there is anything common between those experiencing this problem.

Having one disconnect per day is unacceptable and I consider that to be something very wrong.
But, having one disconnect a week, as annoying as it is, is not uncommon, and I don't believe that will change by switching the xDrip release.

The intermittent connectivity problem is not new.  This page has existed for quite a very long time:   https://navid200.github.io/xDrip/docs/Intermittent.html

So far, this thread is showing that there is no issue with recent releases of xDrip. 

If you find out what the average disconnect rate is and then switch back to an older version and you get a lower rate, please let us know.

**Reply by marinabjezancevic** [reply]
Thank you for your assistance an guiding. For the time being I will switch to an older version because the Xdrip is for my child in daycare and I need it to be without disconnections as much as possible. Best regards :)

**Reply by jamorham** [reply]
If you get problems like occasionally/randomly needing to perform pairing again you can try the `special pairing workaround` option. I don't think changing xDrip version will make any difference for this problem as it is more of a low-level bluetooth problem.

**Reply by marinabjezancevic** [reply]
![1682005680102](https://user-images.githubusercontent.com/130890697/233419369-66494118-d414-4075-ab03-68a6d6b32a03.jpg)
 
Sometimes it asks for pairing, but other times just keeps being disconnected. Special pairing workaround is enabled.

**Reply by jamorham** [reply]
I think unless there are very obvious problems with it, you should use the latest nightly instead of that older version. Any bug fixes etc are more likely to be in the recent nightly versions.

**Reply by marinabjezancevic** [reply]
It seems to be that the problem was in Android 11. 
I followed additional settings from dontkillmyapp webpage, and now the connection is stable. 
Now using the latest update.

**Reply by jamorham** [reply]
@marinabjezancevic are you able to provide a link to the instructions you followed so that it may help others. thanks

**Reply by marinabjezancevic** [reply]
Sure, Navid provided me link to that page, and I just followed recommended settings for Samsung Android 11:

https://navid200.github.io/xDrip/docs/Intermittent.html 

https://dontkillmyapp.com/samsung

**Comment by miglesmigle** [comment]
Phone: Real ME
![image](https://user-images.githubusercontent.com/131362494/233355300-2c9716ef-f4a6-4012-9e8f-adff783c609d.png)
Android 11
![image](https://user-images.githubusercontent.com/131362494/233354937-75b81c33-9c54-43f6-83fc-81757bebecee.png)
Couldn't connect to C transmitter using January 18th, 2023 xDrip version (received a message "Problem downloading plugin"):
![image](https://user-images.githubusercontent.com/131362494/233355577-1b5f4083-f74f-4d87-be63-7ea476e1ee37.png)
Installed xDrip April 17th, 2023 version - transmitter connected successfully. No complaints after that (for > 24 hours).

**Comment by reyalex1** [comment]
Currently using 2023.02.26 with pixel 6 running android 13. Using Dexcom One transmitter starting with C. If lucky can go a couple of days without pair request but mostly have 2/3 per day. Have logged this previously so think you're aware, but not attempted an update since things are working otherwise. Should I update and report findings on current version? Can I roll back if there's any more major issue?

**Reply by jamorham** [reply]
Yes you can roll back as needed, just install whichever xDrip version you want over the top of what you are using. @reyalex1 do you have the `Special Pairing Workaround` enabled?

**Reply by reyalex1** [reply]
Thanks for confirming.

I do not have special pairing workaround enabled as had presumed it applied only to Samsung's. Unless suggest otherwise, will enable now and test before any update to isolate the change in case it works for me.

**Reply by jamorham** [reply]
I've heard that it can resolve the issue of occasionally losing pairing and it being asked for again. See if it works for you.

**Reply by reyalex1** [reply]
No pair request since enabling, about 24 hours in. I will report back if this continues for a few days as it would be out of the norm. Have not updated version.

**Reply by reyalex1** [reply]
Still no more pair requests since enabling special pairing workaround, for I think 3.5 days now. So my issues seem to be fixed 😀

Using 2023.02.26.

**Comment by Dragana2007** [comment]
We have a transmitter starting with C and used all drip releases from 15.3.2023. to 17.4.2023. We had the same problem with each one, they would lose the signal very quickly which would come back if I changed the last letter of the transmitter and retyped the correct one. Yesterday we installed xdrip release from 22.04.2023. and it works for now. 
The phone is Huawei p20lite, android 9.

**Reply by Dragana2007** [reply]
EDIT: release 22.04.2023. worked for a couple of days and then problem appeared as in earlier versions.

**Comment by MarkoJ1974** [comment]
In the 20 days I use dexcom One, I went trough all the Nigthy versions and
didn't have any issues.

**Reply by Dragana2007** [reply]
What phone do you have

**Comment by MarkoJ1974** [comment]
Samsung S22 ultra

**Comment by tommasavas** [comment]
Hello
I have transmiter C235MC. I tryed it with a lot of samsung phones A07, A14, A20e, A22 5G with no succes. Before this transmiter I had another transmiter with C in the begining and it was working. Attached log from xdrip. 
![Uploading Screenshot_20230501_215735_xDrip+.jpg…]()

**Reply by Navid200** [reply]
You can see how you can attach a screenshot if you look at the welcome message here: https://github.com/NightscoutFoundation/xDrip/discussions/1900

For Samsung, in addition to disabling battery optimization, you also need to disable the setting for putting unused apps to sleep.  I have explained it here: https://navid200.github.io/xDrip/docs/Intermittent.html

**Reply by tommasavas** [reply]
Phone connects to transmiter, but after 5 min disconects always. Okey, i didn't try to turn off avoid scaning. I will try it. 
Log:
![image](https://user-images.githubusercontent.com/74055596/235516271-4f4cd700-37fa-4622-bc3c-49720a378f9f.png)

**Reply by Navid200** [reply]
Are you sure you have enabled the "Special pairing workaround" setting?

**Reply by tommasavas** [reply]
Yes I'm sure. I try diferent settings... no succes. 
Avoid scaning - no succes. Disconected after 5 min.

**Reply by tommasavas** [reply]
I had few sensors with C, i havn't issue like  with this transmiter.

**Reply by tommasavas** [reply]
If you want I can sent transmiter to you.

**Reply by Navid200** [reply]
If I receive your transmitter and it happens to work fine for me, it will not prove anything.

Do you mind posting a screenshot of the following page?
Settings &#8722;> G5/G6/Dex1 Debug Settings

**Reply by tommasavas** [reply]
Debug settings:
![image](https://user-images.githubusercontent.com/74055596/235528140-0e95ed8c-f65d-4d1d-a8c5-f8d06343bf64.png)

**Reply by Navid200** [reply]
What version of xDrip are you running?  Is it the same version that generated those logs you have posted?

**Reply by tommasavas** [reply]
Log was from 04.24, screen 02.20. Same configurations on 04.24. Trying everithing. This night I tryed original dex app, I got disconnected like with xdrip. However with dex receiver no problems.

**Reply by tommasavas** [reply]
I think, that it can be broke transmiter from the factory.

**Reply by Navid200** [reply]
You can only conclude that it is a broken transmitter if you have a Dexcom receiver or a phone that is on their approved list.

**Comment by Dragana2007** [comment]
Since we had problems with xdrip with previous phone, we bought a new phone and xdrip release from 24.4.2023. and works great. The phone is Cubot King Kong mini 2 pro 4/64 GB, android 11.

**Comment by PeterCGM** [comment]
Hello,
I have a dex 1 transmitter C24K9T, tried 2 phones: Xiomi Redmi Note 8T (Android 11) and Motorola E40 (Android 11) and tried all possible settings combinations mentioned above, as well as many xDrip+ versions up to May 2023. Just after the first pairing request follows "Unpaired" - same on both phones. Now the only viable option seems to be a Dex 1 Receiver but we loose remote monitoring.
I am totally lost ...

**Reply by Navid200** [reply]
Use the stable release of xDrip.   https://github.com/NightscoutFoundation/xDrip/releases/tag/2023.05.03

Use these settings:  https://navid200.github.io/xDrip/docs/G6-Recommended-Settings.html  
Don't change the settings.
  
Enable Special pairing workaround.

**Reply by PeterCGM** [reply]
Hi Navid200, many thanks for the prompt response. I followed this step by step and then double checked every setting.
Unfortunately every second connect request returns "Unpaired"

![Screenshot_dexdrip](https://github.com/NightscoutFoundation/xDrip/assets/71789897/d4ce4d84-a8fa-470c-841c-81bc70baca85)
![Screenshot_dexdrip1](https://github.com/NightscoutFoundation/xDrip/assets/71789897/0e36a081-e586-47a0-bf07-3d2e6c4c2375)

**Reply by Navid200** [reply]
Please post a screenshot of the following page:
Settings &#8722;> G5/G6/Dex1 Debug Settings

**Reply by PeterCGM** [reply]
Here it is

![Screenshot_dexdrip2](https://github.com/NightscoutFoundation/xDrip/assets/71789897/4c84f144-c8f0-41c8-8975-07bc8e98a628)
![Screenshot_dexdrip3](https://github.com/NightscoutFoundation/xDrip/assets/71789897/9c43cc4b-10d3-4daf-a46d-dc1e1382377f)
last option on second page is not selected (Unbond G5 before each read)

**Reply by Navid200** [reply]
Please deselect "Avoid  Scanning".
Does that make a difference?

**Reply by PeterCGM** [reply]
No doesn't make a difference, still second connect request returns "Unpaired"

I am wondering why OB1 collection service says:  14:46 removingBond: D2:4B:2C:51:26:BF
 
![Screenshot_dexdrip4](https://github.com/NightscoutFoundation/xDrip/assets/71789897/6a59fdc3-5183-42bd-9856-06326cfe513c)


https://github.com/NightscoutFoundation/xDrip/discussions/2655
From logs perspective this thread looks very similar to my issue, especially about the failing second connect request. However, the difference is that I am not able to establish even a single stable connection. The author assumes some security feature disconnecting the phone if not in range for a while, but I don't believe it. Everyone looses connection for one reason or another and it would be catastrophic to all.

**Reply by Navid200** [reply]
@PeterCGM 
Please create a new discussion.
In it, explain the problem and include your screenshots.  I will then ask a developer who knows more about this to look at it and advise.

**Comment by dar-kod** [comment]
@Navid200 I'm using the Xiaomi Poco F3 and my transmitter starts with C. I had connection issues with the March version of xDrip+, but after figuring out how to turn off MIUI optimization in the developer options, I only have occasional disconnections, which I understand is due to Android 13. To turn off MIUI optimization, you have to tap "Reset to defaults" a few times until "Turn on MIUI optimization" appears, and then you can turn it off.

I think this means that whoever has a problem first needs to check if the [optimization](https://dontkillmyapp.com/xiaomi) is turned off for a specific phone. https://dontkillmyapp.com may be very useful.

**Comment by Raminta-ltu** [comment]
@Navid200 I've got transmitter starting with C and it was working until yesterday for 60 days. Today I was trying to fix the issue - reniewed xdrip to the latest version and used all the suggested setting, but still not success. I'm not sure if transmitter can get broken after 60 days :/ 

Maybe you could share a file with setting which I can upload to xdrip app?

**Reply by tommasavas** [reply]
Probably your transmiter dead. Call dexcom.

**Reply by Navid200** [reply]
@Raminta-ltu Why didn't you open a new thread?  You can start a new thread by clicking on new discussion?
Otherwise, everyone in this thread will get a notification for every post.

Please start a new thread and include a screenshot of the G5/G6 system status page.

**Comment by claytonjwalsh** [comment]
Works fine. Samsung S8 SM-G950U1. Android 9. Xdrip 2024.02.14
